### PR TITLE
FLA compatibility - possible different file ID and world map sector limits

### PR DESCRIPTION
**Pull request by fastman92**
Compatibility with different world sector limits & ID limits.

This provides a possibility for heavy modding, for someone who would add FLA (fastman92 limit adjuster) loading into MTA.

Works properly with limits changed (FLA 5.7+)/limits unchanged.

Sample part of log from the FLA:
---------------------------------------------
At least one ID limit is changed. ID limits will be patched.
Count of file IDs is over 32767 and requires unsigned ID. Applying unsigned ID patches.
Count of file IDs is over 65532 and base ID of CStreamingInfo requires int32_t. Applying int32_t base ID patches.
Count of DFF/TXD IDs is over 65532 and requires int32_t. Applying int32_t DFF+TXD ID patches.
COL ID limit is over 256 and requires something more than uint8_t. Applying COL ID uint32_t patches.
IPL ID limit is over 256 and requires something more than uint8_t. Applying IPL ID uint32_t patches.

New ID limits:
0 - 599999 (600000) - DFF models defined within IDE files
600000 - 1099999 (500000) - TXD texture archives.
1100000 - 1599999 (500000) - COL collision archives.
1600000 - 1600299 (300) - IPL Binary IPL files.
1600300 - 1600363 (64) - DAT files limited to nodes*.dat
1600364 - 1600763 (400) - IFP animation archives.
1600764 - 1601263 (500) - RRR car recordings, carrec*.rrr files
1601264 - 1601345 (82) - SCM scripts
1601346 - 1601347 (2) - Loaded list
1601348 - 1601349 (2) - Requested list
1601350 - count of all file IDs

---------------------------------------------
Renderware world map size set to 28000 x 28000
World map size set to 24000 x 24000

World map info info:
World map size: 24000 x 24000

World sectors:
CWorld::ms_aSectors
type: buildings, dummies
count: 480 * 480 = 230400
sector area: 50 x 50

CWorld::ms_aLodPtrLists
type: lod
count: 120 * 120 = 14400
sector area: 200 x 200

CWorld::ms_aRepeatSectors
type: vehicles, peds, object
count: 16 * 16 = 256

----------------

FLA is not an official part of MTA at the moment. If you want to add it, then download MTA source code, edit Client/core/Core.cpp, function DllMain:

Replace the following:
```cpp
            g_hModule = hModule;
            g_pCore = new CCore;

            FileTranslator.GetGTARootDirectory(WorkingDirectory);
            SetCurrentDirectory(WorkingDirectory.c_str());
```

With:
```cpp
            g_hModule = hModule;
            g_pCore = new CCore;

            FileTranslator.GetGTARootDirectory(WorkingDirectory);
            SetCurrentDirectory(WorkingDirectory.c_str());

            // Load the FLA
            LoadLibraryA("$fastman92limitAdjuster.asi");
```

Compile the new core.dll file and run.

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp [hunk 1]
```diff
@@ -1608,7 +1610,7 @@ int CLuaEngineDefs::EngineGetModelPhysicalPropertiesGroup(lua_State* luaVM)
 
     if (!argStream.HasErrors())
     {
-        if (iModelID < 0 || iModelID >= 20000)
+        if (iModelID < 0 || iModelID >= GetBaseIDforTXD())
         {
             argStream.SetCustomError("Expected model ID in range [0-19999] at argument 1");
```
Comment (AlexTMjugador) [inline Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: This error message still uses a hardcoded number. I think the intent here is to derive the number from the base TXD ID.

```suggestion
        if (iModelID < 0 || iModelID >= GetBaseIDforTXD())
        {
            argStream.SetCustomError(SString("Expected model ID in range [0-%d] at argument 1", GetBaseIDforTXD() - 1));
```

Comment (fastman92) [reply Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: Right. I'll alter it.

Comment (fastman92) [reply Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp]: Done. Now it's all complete. New commit submitted. It's all made to work.

**Comment by fastman92** [comment]
CClientGame::RestreamWorld for value 26316, which is total, ount of file IDs still remains to be changed. I'll do it today later.

**Comment by ghost** [comment]
~~Thanks for the PR, but I don't think we can merge this. In MTA, we don't provide compatibility support to forks/mods. The main reason is we're actively updating MTA, and the source code is modified/refactored very often. Eventually, your mod will no longer work on the latest version of MTA. So we ask people to implement their features directly into MTA. Like most successful forks out there, you can also maintain your own fork and update it from time to time.~~

**Comment by Pirulax** [comment]
The title is a little misleading, as the changes in this PR are rather generic, eg.: changing short to int, which will come handy in the future for as well.
Also 1-2 possible crash fixes.

**Comment by ghost** [comment]
I had a discussion about your PR with the rest of the team. The conclusion was that your changes are good for MTA, although some things could've been done differently. There are some improvements needed, and I'll improve your PR to make sure it doesn't violate MTA's coding guidelines. I mentioned how your mod may no longer work with MTA in the future. We definitely will try not to cause any breakage, but there's a possibility that it can happen.

> 
> 
> The title is a little misleading, as the changes in this PR are rather generic, eg.: changing short to int, which will come handy in the future for as well.
> Also 1-2 possible crash fixes.

Yes, I agree.

**Review by ghost — APPROVED** [review]

**Pull request merged**